File: leetcode/435.non-overlapping-intervals/ts/eraseoverlapintervals.ts
const eraseOverlapIntervals = (intervals: number[][]): number => {
    if (intervals.length < 2) return 0

    intervals.sort((a, b) => {
        if (a[1] !== b[1]) return a[1] - b[1]
        return a[0] - b[0]
    })

    let prevIdx = 0
    let res = 0

    for (let i = 1; i < intervals.length; i++) {
        if (intervals[i][0] < intervals[prevIdx][1]) {
            res++
            continue
        }

        prevIdx = i
    }

    return res
}

File: leetcode/435.non-overlapping-intervals/cpp/eraseoverlapintervals.cpp
#include <vector>

class Solution
{
public:
    int eraseOverlapIntervals(std::vector<std::vector<int>> &intervals)
    {
        if (intervals.size() < 2)
            return 0;

        std::sort(intervals.begin(), intervals.end(),
                  [](std::vector<int> a, std::vector<int> b)
                  {
                      if (a[1] != b[1])
                          return a[1] < b[1];
                      return a[0] < b[0];
                  });

        int res = 0;
        int prevIdx = 0;

        for (int i = 1; i < intervals.size(); i++)
        {
            if (intervals[i][0] < intervals[prevIdx][1])
            {
                res++;
                continue;
            }

            prevIdx = i;
        }

        return res;
    }
};
